cmake/nuttx_mkconfig.cmake

```diff
@@ -91,13 +91,21 @@ file(APPEND ${CONFIG_H}
      "/* General Definitions ***********************************/\n")
 file(APPEND ${CONFIG_H} "#define CONFIG_BASE_DEFCONFIG \"${BASE_DEFCONFIG}\"\n")
 
-file(STRINGS ${CMAKE_BINARY_DIR}/.config ConfigContents)
-foreach(NameAndValue ${ConfigContents})
+function(process_individual_rows)
+  set(NameAndValue ${ARGN})
+  list(LENGTH NameAndValue LENGTH_RESULT)
```

Comment:
let's keep the naming style consistent (NameAndValue, LENGTH_RESULT, configFileContent).